### PR TITLE
Add Artifact Details tab bar

### PR DESCRIPTION
**Pull request by kwasi**
Note: A follow-up change will push the tab route to the history stack.

Screenshot: https://screenshot.googleplex.com/VF7AgsBKjMM
Mock: https://gallery.googleplex.com/projects/MCHbtQVoQ2HCZV4FTvhm3v2g/files/MCEJu8Y2hyDSccyl-plWHUJtkG6GfbwEzBU

/area front-end
/kind feature
/assign @kwasi
/cc @avdaredevil @jessiezhu @renming

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/metadata/139)

### CODE REVIEW

File: frontend/src/atoms/__snapshots__/MD2Tabs.test.tsx.snap [hunk 1]
```diff
@@ -0,0 +1,351 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Input gracefully handles an out of bound selectedTab value 1`] = `
```
Comment (avdaredevil) [inline frontend/src/atoms/__snapshots__/MD2Tabs.test.tsx.snap]: What is this file @kwasi ?

File: frontend/src/pages/ArtifactDetails.test.tsx [hunk 2]
```diff
@@ -1,42 +1,41 @@
 import * as React from 'react';
-import ArtifactDetails from './ArtifactDetails';
-import {PageProps} from './Page';
-import {shallow, ShallowWrapper, ReactWrapper} from 'enzyme';
-import {Api} from '../lib/Api';
-import * as TestUtils from '../TestUtils'
-import {RouteParams} from '../components/Router';
-import {ApiGetArtifactResponse} from '../apis/service';
+import ArtifactDetails, { ArtifactDetailsTab } from './ArtifactDetails';
+import { PageProps } from './Page';
```
Comment (avdaredevil) [inline frontend/src/pages/ArtifactDetails.test.tsx]: Let's try to stick to the style guide already being used by the codebase?

File: frontend/src/pages/ArtifactDetails.tsx [hunk 3]
```diff
@@ -14,38 +14,56 @@
...
-    this.state = {};
+    this.state = {
+      selectedTab: ArtifactDetailsTab.OVERVIEW
+    };
     this.load = this.load.bind(this);
   }
 
   private get fullTypeName(): string {
-    return this.props.match.params[RouteParams.ARTIFACT_TYPE] || '';
+    return this.props.match.params[RouteParams.ARTIFACT_TYPE] || "";
```
Comment (avdaredevil) [inline frontend/src/pages/ArtifactDetails.tsx]: The preferred JS convention is single-quotes, we use the same practice in Kubeflow/kubeflow
```suggestion
    return this.props.match.params[RouteParams.ARTIFACT_TYPE] || '';
```

File: frontend/src/pages/ArtifactDetails.tsx [hunk 4]
```diff
@@ -14,38 +14,56 @@
...
   }
 
   private get properTypeName(): string {
-    const parts = this.fullTypeName.split('/');
-    if (!parts.length) return '';
+    const parts = this.fullTypeName.split("/");
+    if (!parts.length) return "";
```
Comment (avdaredevil) [inline frontend/src/pages/ArtifactDetails.tsx]: Same ^^

File: frontend/src/atoms/__snapshots__/MD2Tabs.test.tsx.snap (hunk 1 above)
Comment (kwasi) [reply frontend/src/atoms/__snapshots__/MD2Tabs.test.tsx.snap]: This repo is using Jest snapshots: https://jestjs.io/docs/en/snapshot-testing

This is a new snapshot for the MD2Tabs component.

https://github.com/kubeflow/metadata/tree/master/frontend/src/atoms/__snapshots__

File: frontend/src/pages/ArtifactDetails.test.tsx (hunk 2 above)
Comment (kwasi) [reply frontend/src/pages/ArtifactDetails.test.tsx]: Reverted these changes in the latest version. I added Prettier locally but it wasn't following the existing style. I can add a config that uses the current styles in a separate PR.

File: frontend/src/pages/ArtifactDetails.tsx (hunk 3 above)
Comment (kwasi) [reply frontend/src/pages/ArtifactDetails.tsx]: Done.

File: frontend/src/pages/ArtifactDetails.tsx (hunk 4 above)
Comment (kwasi) [reply frontend/src/pages/ArtifactDetails.tsx]: Done.

**Comment by kwasi** [comment]
PTAL ready for review

**Comment by avdaredevil** [comment]
/lgtm
/approve

Thanks, Kwasi!

File: frontend/src/atoms/MD2Tabs.tsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+            onClick={() => { if (i !== selected) { switchHandler(i); } }}>
+            <span ref={this._tabRefs[i]}>{tab}</span>
+          </Button>
+        ))}
+      </div>
+    );
+  }
+
+  public componentDidMount(): void {
+    this._timeoutHandle = setTimeout(this._updateIndicator.bind(this));
```
[the remark this answers is not in the document]
Comment (kwasi) [reply frontend/src/atoms/MD2Tabs.tsx]: Done

**Comment by kwasi** [comment]
Thanks for the review. I decided to go ahead and make the small change now.

**Pull request merged**